firmware/drivers/citiroc/citiroc.c

```diff
@@ -87,6 +87,27 @@ void citiroc_daq_set_hvps_curr(uint16_t curr)
 	CITIROC->HVPSR |= ((curr & 0xffff)<<16);
 }
 
+void citiroc_calib_set_calibration(uint32_t calibration)
+{
+	uint32_t temp;
+	temp = calibration & 0x7fffffff;
+
+	if (temp != 0)  //For frequencies greater than zero, calculate clock divider value
+	{
+		temp = 50000000 / temp;
```

Comment:
The interface looks a bit clunky. Why do you need to divide by 50M here? A peek into the VHDL my reveal some things and I may be able to offer some advice on the implementation. For now it seems that the calibration module could be implemented slightly differently, to avoid this seemingly unnecessary division.

Reply:
The division is needed in order to determine what value the counter in VHDL has to count up to before changing the calibration clock state. As there are two states for the clock '1' and '0', and the internal clock is set to 100Mhz, 50M/frequency will determine the value that has to be counted before switching state.

As an example, if we want to have a 1 Hz clock, 50M/1 will give us 50M as counter value, meaning that the calibration clock will switch states every 0.5 second as the internal clock is set to 100Mhz. 

I have attached the VHDL in Slack, for clairifcation.

Reply:
OK, that makes sense. As I mentioned in the `cubes-gw` review on Slack, changing the interface of the VHDL module to expect a period instead of a frequency makes the module easier to use, avoiding "costly" division operations in the code and more straightforward to understand.